daisy/snapshot_test.go

```diff
@@ -26,12 +26,29 @@ import (
 func TestSnapshotPopulate(t *testing.T) {
 	w := testWorkflow()
 
+	currentTestProject := "currenttestproj"
+	currentTestZone := "currenttestzone"
+	currentTestDisk := "currenttestdisk"
+
 	tests := []struct {
-		desc      string
-		ss        *Snapshot
-		shouldErr bool
+		desc                string
+		ss                  *Snapshot
+		shouldErr           bool
+		expectedExtendedURL string
+		expectedLink        string
 	}{
-		{"default case", &Snapshot{}, false},
+		{"source disk URI: only name", &Snapshot{
+			Resource: Resource{ExactName: true},
+			Snapshot: compute.Snapshot{Name: testSnapshot, SourceDisk: fmt.Sprintf("aaa")}}, false, "aaa", fmt.Sprintf("projects/%s/global/snapshots/%s", testProject, testSnapshot),
+		},
+		{"source disk URI: with zones", &Snapshot{
+			Resource: Resource{ExactName: true},
+			Snapshot: compute.Snapshot{Name: testSnapshot, SourceDisk: fmt.Sprintf("zones/%v/disks/%v", currentTestZone, currentTestDisk)}}, false, fmt.Sprintf("projects/%v/zones/%v/disks/%v", testProject, currentTestZone, currentTestDisk), fmt.Sprintf("projects/%s/global/snapshots/%s", testProject, testSnapshot),
+		},
+		{"source disk URI: with projects and zones", &Snapshot{
+			Resource: Resource{ExactName: true},
+			Snapshot: compute.Snapshot{Name: testSnapshot, SourceDisk: fmt.Sprintf("projects/%v/zones/%v/disks/%v", currentTestProject, currentTestZone, currentTestDisk)}}, false, fmt.Sprintf("projects/%v/zones/%v/disks/%v", currentTestProject, currentTestZone, currentTestDisk), fmt.Sprintf("projects/%s/global/snapshots/%s", testProject, testSnapshot),
+		},
```

Comment:
I'm generally not a stickler over line length, but here and a few other places the lines are so long that I have to horizontally scroll to read everything

Reply:
wrapped the text in the latest version